src/darwin/CHIPTool/CHIPTool/Framework Helpers/DefaultsUtils.m

```diff
@@ -82,6 +82,18 @@ void CHIPSetNextAvailableDeviceID(uint64_t id)
     return device;
 }
 
+CHIPDevice * GetPairedDeviceWithID(uint64_t deviceId)
+{
+    CHIPToolPersistentStorageDelegate * storage = [[CHIPToolPersistentStorageDelegate alloc] init];
+    dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.persistentstorage.callback", DISPATCH_QUEUE_SERIAL);
+
+    CHIPDeviceController * controller = [CHIPDeviceController sharedController];
+    [controller setPersistentStorageDelegate:storage queue:callbackQueue];
```

Comment:
I feel like I'm missing something here.  Each call to `GetPairedDeviceWithID` will set up a new storage delegate and queue, right?

But then things like `refreshDeviceList` call `GetPairedDeviceWithID` in a loop.  Do we really want to create a new storage delegate for each loop iteration there?

I see that `GetPairedDevice` does some similar things, but I don't know whether it makes more sense to do this in the context there...

Reply:
The main problem is that we don't have a place to initialize the controller object in the CHIPTool iOS app. The controller object is singleton, and when it gets created, the delegate objects are not available till then. We can use some cleanup in this part of the code.